### PR TITLE
[COM-28756]: update scarcity

### PR DESCRIPTION
**Pull request by sizhang12**
Updating scarcity to include rendering the total scarcity amount when `scarcityShownByDefault` is true but also rendering the scarcity with attributes after. 

If `scarcityShownByDefault` is true, then that means the first index of `scarcityTemplateViews` is the total stock. Everything after that are by variants with attributes.

### CODE REVIEW

File: core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html [hunk 1]
```diff
@@ -27,10 +27,10 @@
 {item|product-scarcity}
 
 :OUTPUT
-<div class="product-scarcity" data-variant-attributes="{&quot;Color&quot;:&quot;Red&quot;}" hidden>
-    Only 3 left!
-  </div>
+  <div class="product-scarcity" data-variant-attributes="{&quot;Color&quot;:&quot;Red&quot;}" hidden>
```
Comment (sizhang12) [inline core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html:30]: I am not sure why spacing matters. Some tests are failing and I am not sure why. There might be extra spacing or invisible character I am missing?

Comment (phensley) [reply core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html:30]: The language is whitespace-sensitive, so the output needs to match whitespace as well.

**Comment by phensley** [comment]
@sizhang12 Please update the commit message to include the full description of the change.

File: core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html (hunk 1 above)
Comment (sizhang12) [reply core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html:30]: @phensley I am not sure what I'm missing here. Appreciate any help with this! 

Screen Shot 2023-07-31 at 10 56 46 AM

Comment (phensley) [reply core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html:30]: @sizhang12 I'll take a look shortly

Comment (sizhang12) [reply core/src/test/resources/com/squarespace/template/plugins/platform/f-scarcity-default-hidden.html]: @phensley  thank you! I will be making the same changes into template-engine as well.

**Pull request merged**